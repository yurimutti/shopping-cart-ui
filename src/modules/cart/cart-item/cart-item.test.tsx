import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { CartItem } from "./";
import type { Product } from "@/types/product";

const mockProduct: Product = {
  id: 1,
  name: "Cheeseburger Combo",
  price: 12.99,
  quantity: 2,
};

describe("<CartItem />", () => {
  it("should render the cart item correctly", () => {
    const { container } = render(
      <CartItem item={mockProduct} onRemove={vi.fn()} />
    );

    expect(screen.getByText(mockProduct.name)).toBeInTheDocument();
    expect(
      screen.getByText(`Quantity: ${mockProduct.quantity}`)
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", {
        name: `Remove ${mockProduct.name} from cart`,
      })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onRemove when remove button is clicked", () => {
    const onRemove = vi.fn();

    render(<CartItem item={mockProduct} onRemove={onRemove} />);

    const removeButton = screen.getByRole("button", {
      name: `Remove ${mockProduct.name} from cart`,
    });

    fireEvent.click(removeButton);

    expect(onRemove).toHaveBeenCalledWith(mockProduct.id);
  });
});
