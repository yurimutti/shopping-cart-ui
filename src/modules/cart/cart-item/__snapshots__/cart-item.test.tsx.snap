// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CartItem /> > should render the cart item correctly 1`] = `
<li
  class="flex justify-between items-center bg-white p-4 rounded-lg border border-gray-200 shadow-sm"
  role="listitem"
>
  <div
    class="flex flex-col gap-1"
  >
    <span
      class="font-medium text-red-600"
    >
      Cheeseburger Combo
    </span>
    <span
      class="text-gray-500 text-sm"
    >
      Quantity: 
      2
    </span>
  </div>
  <button
    aria-label="Remove Cheeseburger Combo from cart"
    class="text-red-600 hover:text-red-700 text-sm font-medium"
  >
    Remove
  </button>
</li>
`;
